### PR TITLE
starship: 0.26.5 -> 0.27.0

### PR DESCRIPTION
**Pull request by Br1ght0ne**
###### Motivation for this change
https://github.com/starship/starship/releases/tag/v0.27.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @bbigras @davidtwco

---
This change is [Reviewable](https://reviewable.io/reviews/nixos/nixpkgs/74923)

### CODE REVIEW

**Comment by davidtwco** [comment]
I'm seeing the same test failure as ofborg locally.

**Comment by filalex77** [comment]
@davidtwco I added a patch to fix absolute paths in tests. Should be fine now.
@GrahamcOfBorg build starship

**Review by davidtwco — APPROVED** [review]

**Comment by bbigras** [comment]
I have a dumb question, I'm still new to all of this. There's no other way than patching the tests? It's not possible to make the tests run with coreutils in PATH? To avoid a patch that could break on any release.

**Review by bbigras — APPROVED** [review]
I'm still approving. It builds and runs fine. :+1:

File: pkgs/tools/misc/starship/default.nix [hunk 1]
```diff
@@ -1,26 +1,35 @@
...
   };
 
   buildInputs = stdenv.lib.optionals stdenv.isDarwin [ libiconv darwin.apple_sdk.frameworks.Security ];
 
+  patches = [
+    (substituteAll {
+      src = ./fix-fhs-paths-in-tests.patch;
+      inherit coreutils;
+    })
+  ];
```
Comment (jonringer) [inline pkgs/tools/misc/starship/default.nix]: alternative, as coreutil should be present:
```
  postPatch = ''
    substituteInPlace src/utils.rs \
      --replace "/bin/echo" "echo"
  '';
```

then you can remove the patch

File: pkgs/tools/misc/starship/default.nix [hunk 2]
```diff
@@ -1,26 +1,35 @@
-{ stdenv, fetchFromGitHub, rustPlatform, libiconv, darwin }:
+{ stdenv, fetchFromGitHub, rustPlatform
+, libiconv, darwin
+, substituteAll, coreutils }:
```
Comment (jonringer) [inline pkgs/tools/misc/starship/default.nix]:
```suggestion
}:
```

**Review by bbigras — APPROVED** [review]

**Comment by jonringer** [comment]
@Ma27 the last commit should have been amended

**Pull request merged**